hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestECContainerReplicaCount.java

```diff
@@ -229,7 +229,7 @@ public void testAdditionalMaintenanceCopiesAllMaintenance() {
             Collections.emptyList(), 1);
     Assertions.assertFalse(rcnt.isSufficientlyReplicated(false));
     Assertions.assertFalse(rcnt.isOverReplicated(true));
-    Assertions.assertEquals(4, rcnt.additionalMaintenanceCopiesNeeded());
+    Assertions.assertEquals(4, rcnt.additionalMaintenanceCopiesNeeded(false));
```

Comment:
All the calls I can see in the this test calss call `rcnt.additionalMaintenanceCopiesNeeded(false)` with `false`. We should add at least one test with some pending add and test the `true` condition too.